src/rgw/rgw_sal_motr.cc

```diff
@@ -1196,15 +1196,16 @@ int MotrBucket::list(const DoutPrefixProvider *dpp, ListParams& params, int max,
   int rc;
   if (max == 0)  // Return an emtpy response.
     return 0;
-  max++;  // Fetch an extra key if available to ensure presence of next obj.
-  vector<string> keys(max);
-  vector<bufferlist> vals(max);
+
   string tenant_bkt_name = get_bucket_name(info.bucket.tenant, info.bucket.name);
 
   ldpp_dout(dpp, 20) << __func__ << ": bucket=" << tenant_bkt_name
                     << " prefix=" << params.prefix
                     << " marker=" << params.marker
-                    << " max=" << max - 1 << dendl;
+                    << " max=" << max << dendl;
+  int batch_size = 100;
```

Comment:
we already have logic to fetch keys in batch of 100 in next_query by name.
any reason to repeat it here?

Reply:
As there are extra entries present in the index with format object_name/null for each object, we have to fetch more entries from the motr and filter out the un-wanted entries until the count of the valid entries is equal to `max`. I am try to fetch entries from the mort in the batch of 100.